File: frontend/src/types/workspace.ts
export interface Workspace {
  id: string;
  name: string;
  ownerId: string;
  city: string;
  address: string;
  initials: string;
  totalBeds: number;
  occupiedBeds: number;
  accent: "blue" | "violet" | "emerald" | "rose";
  planId?: "monthly" | "yearly";
  subscriptionStatus?: string;
  stripeSubscriptionId?: string;
  stripeCustomerId?: string;
  createdAt: string;
}

File: backend/app/models/workspace.py
"""Workspace model – shape of a row from the workspaces table."""
from pydantic import BaseModel
from typing import Optional


class WorkspaceModel(BaseModel):
    """Represents a workspace returned by the API (matches _fmt output in workspaces.py)."""
    id: str
    name: str
    ownerId: str
    city: Optional[str] = None
    address: Optional[str] = None
    initials: Optional[str] = None
    totalBeds: int = 0
    occupiedBeds: int = 0
    accent: str = "blue"
    planId: Optional[str] = None
    createdAt: str
